### PR TITLE
feat(keybinds.lua): add `desc` fields to task keybinds

### PR DESCRIPTION
**Pull request by sardonicsloth**
this is to support proper desc= options for which-key, not clear if this is the right place to do this or not. I also don't like setting opts = defaults in the loop but i couldn't think of another way with out changing the rest of the code.

### CODE REVIEW

**Comment by katawful** [comment]
The descriptions feel too ambiguous, hard to know what they are doing. I would suggest adding a verb

File: lua/neorg/modules/core/keybinds/module.lua [hunk 1]
```diff
@@ -316,18 +316,23 @@ module.public = {
...
                                 keymap[1],
-                                "<cmd>Neorg keybind "
-                                    .. mode
-                                    .. " "
-                                    .. table.concat(vim.list_slice(keymap, 2), " ")
-                                    .. "<CR>",
+                                "<cmd>Neorg keybind " .. mode .. " " .. keymap[2] .. "<CR>",
                                 opts
                             )
+                            opts = {
```
Comment (katawful) [inline lua/neorg/modules/core/keybinds/module.lua:332]: Shouldn't this be above `payload.map()`?

Comment (sardonicsloth) [reply lua/neorg/modules/core/keybinds/module.lua:332]: This is a "reset" of the options to the default passed. The original function in keybinds.lua send a list of keybinds that is looped through and then a set of options { silent=true, noremap=true } but it doesn't allow for setting the descriptions per map. This is the best workaround i can think of with out changing everything about the function. Let me know if you want a better way

**Comment by vhyrro** [comment]
Thanks for the PR! I think some extra work needs to be done here in order to make everything work with neorg, but these require some extra changes to the API. The fix to allow `{ desc = ...}` is good on paper, but Neorg reserves this as a "parameters" section for keybinds, where there may be any amount of parameters given. I think a better call would be to allow the table to receive a `opts = {}` field which overrides the opts of the mapping function. An example would be:
```lua
{ leader .. "tu", "core.qol.todo_items.todo.task_undone", opts = { desc = "Undone" } }
```
And then revert the change to the mapping function to use an `opts` field instead.

The descriptions could be more along the lines of `Mark Task as Done`, `Mark Task as Uncertain` and so on. If you don't beat me to it in about half an hour I can work on this a little bit :)

**Comment by vhyrro** [comment]
Beat you to it :p

Thank you for the PR! I'll go on to add descriptions for the rest of the keybinds now :D

**Pull request merged**